src/plugins/discover/public/application/components/context_app/context_app_legacy.tsx

```diff
@@ -17,33 +17,86 @@
  * under the License.
  */
 import React from 'react';
-import { FormattedMessage } from '@kbn/i18n/react';
+import { FormattedMessage, I18nProvider } from '@kbn/i18n/react';
 import { EuiPanel, EuiText } from '@elastic/eui';
-import { I18nProvider } from '@kbn/i18n/react';
 import {
   DocTableLegacy,
   DocTableLegacyProps,
 } from '../../angular/doc_table/create_doc_table_react';
 import { IIndexPattern, IndexPatternField } from '../../../../../data/common/index_patterns';
 import { LOADING_STATUS } from './constants';
+import { ActionBar, ActionBarProps } from '../../angular/context/components/action_bar/action_bar';
 
-export interface ContextAppProps {
+export type ContextAppProps = Partial<ActionBarProps> & {
   columns: string[];
   hits: Array<Record<string, unknown>>;
   indexPattern: IIndexPattern;
   filter: (field: IndexPatternField | string, value: string, type: '+' | '-') => void;
   minimumVisibleRows: number;
   sorting: string[];
   status: string;
+  predecessorCount: number;
+  successorCount: number;
+  predecessorAvailable: number;
+  successorAvailable: number;
+  onChangePredecessorCount: (count: number) => void;
+  onChangeSuccessorCount: (count: number) => void;
+  predecessorStatus: string;
+  successorStatus: string;
+};
+
+const PREDECESSOR_TYPE = 'predecessors';
+const SUCCESSOR_TYPE = 'successors';
+
+function isPredecessorType(type: string) {
```

Comment:
Since `isPredecessorType` is only used once, you could simplify by migrate it to the `actionBarProps` like `const isPredecessorType = isPredecessorType;`

Reply:
I'm sure you won't do it like this `const isPredecessorType = isPredecessorType;` 🤦